addons/fetchmail_gmail/models/fetchmail_server.py

```diff
@@ -1,44 +1,48 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, _
-from odoo.exceptions import UserError
+from odoo import api, fields, models
 
 
 class FetchmailServer(models.Model):
     _name = 'fetchmail.server'
     _inherit = ['fetchmail.server', 'google.gmail.mixin']
 
-    @api.constrains('use_google_gmail_service', 'server_type')
-    def _check_use_google_gmail_service(self):
-        if any(server.use_google_gmail_service and server.server_type != 'imap' for server in self):
-            raise UserError(_('Gmail authentication only supports IMAP server type.'))
+    server_type = fields.Selection(selection_add=[('gmail', 'Gmail OAuth Authentication')], ondelete={'gmail': 'set default'})
 
-    @api.onchange('use_google_gmail_service')
-    def _onchange_use_google_gmail_service(self):
+    @api.onchange('server_type', 'is_ssl', 'object_id')
+    def onchange_server_type(self):
         """Set the default configuration for a IMAP Gmail server."""
-        if self.use_google_gmail_service:
+        if self.server_type == 'gmail':
             self.server = 'imap.gmail.com'
-            self.server_type = 'imap'
             self.is_ssl = True
             self.port = 993
         else:
-            self.server_type = 'pop'
-            self.is_ssl = False
             self.google_gmail_authorization_code = False
             self.google_gmail_refresh_token = False
             self.google_gmail_access_token = False
             self.google_gmail_access_token_expiration = False
+            super(FetchmailServer, self).onchange_server_type()
 
     def _imap_login(self, connection):
         """Authenticate the IMAP connection.
 
         If the mail server is Gmail, we use the OAuth2 authentication protocol.
         """
         self.ensure_one()
-        if self.use_google_gmail_service:
+        if self.server_type == 'gmail':
             auth_string = self._generate_oauth2_string(self.user, self.google_gmail_refresh_token)
             connection.authenticate('XOAUTH2', lambda x: auth_string)
             connection.select('INBOX')
         else:
             super(FetchmailServer, self)._imap_login(connection)
+
+    def _get_connection_type(self):
```

Comment:
Shouldn't we instead force server_type to imap like previously ? Not sure to understand the change ... you may end up with strange configuration ?

Reply:
Ah ok you added a server_type, nevermind.